include/erlcloud_aws.hrl

```diff
@@ -42,7 +42,7 @@
           secret_access_key::string()|undefined|false,
           security_token=undefined::string()|undefined,
           %% epoch seconds when temporary credentials will expire
-          expiration :: pos_integer(),
+          expiration=undefined :: pos_integer()|undefined,
```

Comment:
I agree with you that non explicit setting is more laziness and just making dializer happy without making code cleaner and thus also prefer explicit setting. 

let's leave it here as is.